src/Umbraco.Web/UmbracoComponentRenderer.cs

```diff
@@ -180,7 +180,7 @@ internal IHtmlString RenderMacro(MacroModel m, IDictionary<string, object> param
                     _umbracoContext.HttpContext.Response.ContentType = contentType;
 
                     //Now, we need to ensure that local links are parsed
-                    html = TemplateUtilities.ParseInternalLinks(output.ToString());
+                    html = TemplateUtilities.ParseInternalLinks(output.ToString(), UmbracoContext.Current.UrlProvider);
```

Comment:
In v8 we are trying to reduce as many singleton usages as possible and use DI for almost everything. This class is already wired up for DI so there is an existing `_umbracoContext` property available. I've fixed this up.

Reply:
Good point! Thanks for fixing! 👍